cmd/list/cluster/cmd.go

```diff
@@ -39,15 +40,35 @@ var Cmd = &cobra.Command{
 	Run:  run,
 }
 
-var listAll bool
+const clusterCount = 1000
+
+var args struct {
+	listAll        bool
+	accountRoleArn string
+}
 
 func init() {
 	flags := Cmd.Flags()
 	flags.SortFlags = false
 
 	output.AddFlag(Cmd)
-	flags.BoolVarP(&listAll, "all", "a", false, "List all clusters across different AWS "+
+	flags.BoolVarP(&args.listAll, "all", "a", false, "List all clusters across different AWS "+
 		"accounts under the same Red Hat organization")
+	flags.StringVar(&args.accountRoleArn, "account-role-arn", "", "List all clusters "+
+		"using the account role identified by the ARN")
+}
+
+func listClustersUsingAccountRole(creator *aws.Creator, runtime *rosa.Runtime) ([]*v1.Cluster, error) {
+	role, err := runtime.AWSClient.GetAccountRoleByArn(args.accountRoleArn)
+	if err != nil {
+		return []*v1.Cluster{}, err
+	}
+
+	if role == nil {
```

Comment:
@oriAdler I removed this check in favour of `GetAccountRoleByArn` returning an error if the role does not exist.

If the role doesn't have tags, then it is still returned from the `mapToAccountRole` function, but the `Policy` fields are not filled. I've updated it so that the `RoleName` and `RoleARN` fields are always completed, which is a slight change to existing behaviour,  but from what I can see, that shouldn't be a problem.